Refreshing the build

TonioGela opened the pull request:
I did some housekeeping, mostly updating the dependencies and switching to sbt-typelevel.

I would like to also add support for Mergify, since there's not that much of a need for a manual intervention to merge PRs on this repo as long as the tests are green.

As there's already support for 2.13 and SJS:

Closes #141.

This also closes all the other Scala Steward PRs, but there's no way to mark them automatically as superseded.

TonioGela committed:
Refreshing the build

TonioGela commented:
@armanbilge I just noticed that the diff is huge due to the change of scalafmt version. If you wish I can split the scalafmt change in a separate commit and add a `.git-blame-ignore-revs`.

armanbilge commented:
> f you wish I can split the scalafmt change in a separate commit and add a `.git-blame-ignore-revs`.

Yes, that's always good practice!

TonioGela committed:
Updating scalafmt configuration

TonioGela committed:
Adding commit to .git-blame-ignore-revs

TonioGela commented:
> > f you wish I can split the scalafmt change in a separate commit and add a `.git-blame-ignore-revs`.
> 
> Yes, that's always good practice!

Hmm, I'm not sure I did it right. I updated the formatter in `f07386a3e982d36d6fc373162292ef1dd3feaa61` and added a `.git-blame-ignore-revs` file with that hash as content, but the diff is still showing the reformatting 🤔
